stratum/hal/lib/common/yang_parse_tree_paths.cc

```diff
@@ -3579,28 +3644,40 @@ void YangParseTreePaths::AddSubtreeInterfaceFromOptical(
 
   node = tree->AddNode(
       GetPath("components")("component", name)("state")("type")());
-  SetUpComponentsComponentStateType(name, "OPTICAL_CHANNEL", node);
+  SetUpComponentsComponentStateType("OPTICAL_CHANNEL", node);
 }
 
 void YangParseTreePaths::AddSubtreeNode(const Node& node, YangParseTree* tree) {
   // No need to lock the mutex - it is locked by method calling this one.
-  TreeNode* tree_node = tree->AddNode(GetPath("debug")("nodes")(
-      "node", node.name())("packet-io")("debug-string")());
+  const std::string& name = !node.name().empty()
+                                ? node.name()
+                                : absl::StrFormat("node-%d", node.id());
+  TreeNode* tree_node = tree->AddNode(
+      GetPath("debug")("nodes")("node", name)("packet-io")("debug-string")());
   SetUpDebugNodesNodePacketIoDebugString(node.id(), tree_node, tree);
-  tree_node = tree->AddNode(GetPath("components")("component", node.name())
-      ("integrated-circuit")("config")("node-id")());
-  SetUpComponentsComponentIntegratedCircuitConfigNodeId(node.id(),
-                                                        tree_node, tree);
-
-  tree_node = tree->AddNode(GetPath("components")("component", node.name())
-      ("integrated-circuit")("state")("node-id")());
-  SetUpComponentsComponentIntegratedCircuitStateNodeId(node.id(),
-                                                       tree_node, tree);
+  tree_node = tree->AddNode(GetPath("components")(
+      "component", name)("integrated-circuit")("config")("node-id")());
+  SetUpComponentsComponentIntegratedCircuitConfigNodeId(node.id(), tree_node,
+                                                        tree);
+  tree_node = tree->AddNode(GetPath("components")(
+      "component", name)("integrated-circuit")("state")("node-id")());
+  SetUpComponentsComponentIntegratedCircuitStateNodeId(node.id(), tree_node,
+                                                       tree);
+  tree_node = tree->AddNode(
+      GetPath("components")("component", name)("state")("type")());
+  SetUpComponentsComponentStateType("INTEGRATED_CIRCUIT", tree_node);
+  tree_node = tree->AddNode(
+      GetPath("components")("component", name)("state")("part-no")());
+  SetUpComponentsComponentStatePartNo(node.id(), tree_node, tree);
+  tree_node = tree->AddNode(
+      GetPath("components")("component", name)("state")("mfg-name")());
+  SetUpComponentsComponentStateMfgName(node.id(), tree_node, tree);
 }
 
 void YangParseTreePaths::AddSubtreeChassis(const Chassis& chassis,
                                            YangParseTree* tree) {
-  const std::string& name = chassis.name();
+  const std::string& name =
+      !chassis.name().empty() ? chassis.name() : "chassis";
```

Comment:
I could not find a unique id in the Chassis proto message. Might be fine, since most configs have this name and there is only on chassis per yang tree.

Reply:
looks good to me